marshmallow_jsonschema/base.py

```diff
@@ -287,21 +378,52 @@ def _from_union_schema(self, obj, field) -> dict[str, list[typing.Any]]:
         msg = f"Field {field} is not a supported Union type."
         raise TypeError(msg)
 
-    def _get_python_type(self, field):
+    def _get_python_type(self, field: fields.Field) -> builtins.type:
         """Get python type based on field subclass"""
+        if PYTYPE_KEY in field.metadata:
+            pytype = field.metadata[PYTYPE_KEY]
+            if not isinstance(pytype, type):
+                msg = (
+                    "A python type was not supplied for 'jsonschema_python_type', in "
+                    f"'{field.attribute or field.name or field.__class__.__name__}'"
+                )
+                raise TypeError(msg)
+            if pytype not in PY_TO_JSON_TYPES_MAP:
+                msg = (
+                    f"'{PYTYPE_KEY}' is not a supported python type for dumping in marshmallow_jsonschema. In field "
+                    "'{field.attribute or field.name or field.__class__.__name__}'"
+                )
+                raise UnsupportedValueError(msg)
+            return pytype
+
         for map_class, pytype in MARSHMALLOW_TO_PY_TYPES_PAIRS:
             if issubclass(field.__class__, map_class):
                 return pytype
 
         msg = f"unsupported field type {field!s}"
         raise UnsupportedValueError(msg)
 
+    def _get_value_from_obj_or_metadata(self, field: fields.Field, attr: str) -> None | typing.Any:
+        """
+        Helper function to search for and return an attribute. First checks for a direct attribute, then checks in
+        metadata. If the attribute value is a function, run and return the function output.
+        Returns None if the attribute is not found in either location.
+        """
+        if hasattr(field, attr):
+            value = getattr(field, attr)
+            return value() if callable(value) else value
+        if attr in field.metadata:
+            value = field.metadata[attr]
+            return value() if callable(value) else value
+        return None
+
     def _get_schema_for_field(self, obj, field):
         """Get schema and validators for field."""
-        if hasattr(field, "_jsonschema_type_mapping"):
-            schema = field._jsonschema_type_mapping()
-        elif "_jsonschema_type_mapping" in field.metadata:
-            schema = field.metadata["_jsonschema_type_mapping"]
+        # For backwards compatibility, can still use '_jsonschema_type_mapping' with JSON equivalent Field type.
```

Comment:
[nit] leftover comment?

Reply:
Nah all good, the diff made it look like it was floating and I thought it was a leftover of that other backwards compatibility comment for the metadata 👍